evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -2754,3 +2755,80 @@ def test_pipeline_transform_with_final_estimator(
         ),
     ):
         pipeline.transform(X, y)
+
+
+@patch("evalml.pipelines.components.LogisticRegressionClassifier.fit")
+def test_training_only_component_in_pipeline_fit(mock_fit):
+    X = pd.DataFrame(
+        {
+            "a": [i for i in range(9)] + [np.nan],
+            "b": [i % 3 for i in range(10)],
+            "c": [i % 7 for i in range(10)],
+        }
+    )
+    y = pd.Series([0] * 5 + [1] * 5)
+    pipeline = BinaryClassificationPipeline(
+        {
+            "Imputer": ["Imputer", "X", "y"],
+            "Drop Rows Transformer": [DropRowsTransformer, "Imputer.x", "y"],
+            "Logistic Regression Classifier": [
+                "Logistic Regression Classifier",
+                "Drop Rows Transformer.x",
+                "Drop Rows Transformer.y",
+            ],
+        },
+        parameters={"Drop Rows Transformer": {"indices_to_drop": [0, 9]}},
+    )
+    pipeline.fit(X, y)
+    assert len(mock_fit.call_args[0][0]) == 8
```

Comment:
There's a time when your brains are too big for me.  This is one of them, haha.  Why should this be true if the issue is completed successfully?

Reply:
Hahaha, I think this issue is pretty confusing! But it's 8 here because during `fit` / training time, we evaluate all components, regardless of whether or not they're training_only. So the DropRowsTransformer will drop the two rows during fit --> a total of 8 rows left!